api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java

```diff
@@ -329,4 +329,13 @@ public Provider getProviderByIdentifier(String identifier) {
 		criteria.add(Restrictions.ilike("identifier", identifier, MatchMode.EXACT));
 		return (Provider) criteria.uniqueResult();
 	}
+    @Override
+	public Provider createProviderFromUser(Provider provider) throws DAOException {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Provider.class);
+		criteria.add(Restrictions.eq("identifier", provider.getProviderId()));		
+		if(provider.getPerson() != null) {
+			return (Provider) criteria.uniqueResult();
+		}
+		return provider;
```

Comment:
This doesn't look to me like it "creates" a provider

Reply:
Does it only saves the already created provider my the user,this implements prividerDao i used that code seemingly it saves and returns privider, you can highlight to me thanks @ibacher

Reply:
@dkayiwa  @ibacher  , are we still available for this, To me after looking again into it, it seems almost there is no current deprected methods, thats my perspective however i dont know whether there is any new thing to add  in here